deepchem/models/tensorgraph/models/text_cnn.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Sep 28 15:17:50 2017
+
+@author: zqwu
+"""
+import numpy as np
+import tensorflow as tf
+import copy
+
+from deepchem.metrics import to_one_hot, from_one_hot
+from deepchem.models.tensorgraph.layers import Dense, Concat, SoftMax, \
+  SoftMaxCrossEntropy, BatchNorm, WeightedError, Dropout, BatchNormalization, \
+  Conv1D, MaxPool1D, Squeeze, Stack, Highway
+from deepchem.models.tensorgraph.graph_layers import DTNNEmbedding
+
+from deepchem.models.tensorgraph.layers import L2Loss, Label, Weights, Feature
+from deepchem.models.tensorgraph.tensor_graph import TensorGraph
+from deepchem.trans import undo_transforms
+
+# Common symbols in SMILES, note that Cl and Br are regarded as one symbol
+default_dict = {
+    '#': 1,
+    '(': 2,
+    ')': 3,
+    '+': 4,
+    '-': 5,
+    '/': 6,
+    '1': 7,
+    '2': 8,
+    '3': 9,
+    '4': 10,
+    '5': 11,
+    '6': 12,
+    '7': 13,
+    '8': 14,
+    '=': 15,
+    'C': 16,
+    'F': 17,
+    'H': 18,
+    'I': 19,
+    'N': 20,
+    'O': 21,
+    'P': 22,
+    'S': 23,
+    '[': 24,
+    '\\': 25,
+    ']': 26,
+    '_': 27,
+    'c': 28,
+    'Cl': 29,
+    'Br': 30,
+    'n': 31,
+    'o': 32,
+    's': 33
+}
+
+
+class TextCNNTensorGraph(TensorGraph):
```

Comment:
Could you add a docstring providing a high level description of the purpose of this class?

Reply:
Added a description of model and references.